File: python_version/stack/paired_parentheses/paired_parentheses.py
# paired parentheses
# Write a function, pairedParentheses, that takes in a string as an argument.
# The function should return a boolean indicating whether or not the string has well-formed parentheses.

# You may assume the string contains only alphabetic characters, '(', or ')'.


def paired_parentheses(string):


    return 


print(paired_parentheses("(david)((abby))"))  # -> True
print(paired_parentheses("()rose(jeff"))  # -> False
print(paired_parentheses(")("))  # -> False
print(paired_parentheses("()"))  # -> True
print(paired_parentheses("(((potato())))"))  # -> True
print(paired_parentheses("(())(water)()"))  # -> True
print(paired_parentheses("(())(water()()"))  # -> False
print(paired_parentheses(""))  # -> True
print(paired_parentheses("))()"))  # False

File: java_version/stack/paired_parentheses/paired_parentheses.java
package stack.paired_parentheses;

// paired parentheses
// Write a function, pairedParentheses, that takes in a string as an argument. 
// The function should return a boolean indicating whether or not the string has well-formed parentheses.


// You may assume the string contains only alphabetic characters, '(', or ')'.
public class paired_parentheses {

    public static boolean pairedParentheses(String str) {

        int top = -1;



        return top == -1 ? true : false;
    }


    public static void main(String[] args) {
        System.out.println(pairedParentheses("(david)((abby))")); // -> 1 (true)
        System.out.println(pairedParentheses("()rose(jeff")); // -> 0 (false)
        System.out.println(pairedParentheses(")(")); // -> 0 (false)
        System.out.println(pairedParentheses("()")); // -> 1 (true)
        System.out.println(pairedParentheses("(((potato())))")); // -> 1 (true)
        System.out.println(pairedParentheses("(())(water)()")); // -> 1 (true)
        System.out.println(pairedParentheses("(())(water()()")); // -> 0 (false)
        System.out.println(pairedParentheses("")); // -> 1 (true)
        System.out.println(pairedParentheses("))()")); // -> 0 (false)

    }
}